lib/library.rb

```diff
@@ -0,0 +1,51 @@
+class Library
+  attr_reader :name,
+              :books,
+              :authors,
+              :checked_out_books,
+              :checkout_tally
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @checkout_tally = Hash.new(0)
```

Comment:
protip: you can declare things in here that you use within your class that you don't *have* to expose; if the behavior of your code works, then you're effectively testing it, right?